frontend/src/Utils/ModelFeatureFlagLoader.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import { useFeatureFlag } from '@openshift/dynamic-plugin-sdk';
+import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+
+export type ModelFeatureFlagLoaderProps = {
+  modelFeature: {
+    flag: string;
+    model: {
+      group: string;
+      version: string;
+      kind: string;
+    };
+  };
+  namespace: string;
+};
+
+/**
+ * Component to wrap setting and loading of model feature flag.
+ * @param ModelFeatureFlagLoaderProps which model and feature flag to set for given namespace.
+ * @returns null
+ */
+const ModelFeatureFlagLoader: React.FC<ModelFeatureFlagLoaderProps> = ({ modelFeature, namespace }) => {
+  const [, setFlag] = useFeatureFlag(modelFeature.flag);
+  const [data, loaded] = useK8sWatchResource({
```

Comment:
@florkbr @christianvogt not sure about this function. We just want to see if the model is present on the API, feels kinda heavy to watch for the resource. But I couldn't figure out how to use the `k8sGetResource`

Reply:
Model feature flags should work in conjunction with API discovery. 
cc @spadgett

Reply:
Right, this is unnecessary and won't work for users who don't have permission to watch those resources. In OpenShift console, we evaluate the model flags whenever API discovery completes against the models stored in redux. Here is the logic:

https://github.com/openshift/console/blob/b3d0c052d305a6469cf043132119de8231a10c66/frontend/public/reducers/features.ts#L113-L125